platform/security/sts/security-sts-propertyclaimshandler/src/main/java/org/codice/ddf/security/sts/claims/property/AttributeFileClaimsHandler.java

```diff
@@ -148,16 +130,51 @@ public ProcessedClaimCollection retrieveClaimValues(
   }
 
   @Override
-  public List<String> getSupportedRealms() {
-    return supportedRealms;
+  public List<Serializable> getValues(final String attributeName) {
+    return Optional.ofNullable(systemHighAttributes.get(attributeName))
+        .orElse(Collections.emptyList());
   }
 
-  @Override
-  public String getHandlerRealm() {
-    return realm;
+  public void setAttributeFileLocation(String attributeFileLocation) throws IOException {
+    this.attributeFileLocation =
+        notNull(attributeFileLocation, "attributeFileLocation may not be null");
+    init();
   }
 
-  public void setAttributeFileLocation(String attributeFileLocation) {
-    this.attributeFileLocation = attributeFileLocation;
+  private void setSupportedClaimTypes() {
+    Set<Map.Entry<String, Object>> entries = json.entrySet();
+    Set<String> claims = new HashSet<>();
+    for (Map.Entry<String, Object> entry : entries) {
+      Object value = entry.getValue();
+      if (value instanceof Map) {
+        Set keySet = ((Map) value).keySet();
+        for (Object key : keySet) {
+          claims.add((String) key);
+        }
+      }
+    }
+
+    supportedClaimTypes = claims.stream().map(URI::create).collect(Collectors.toList());
+  }
+
+  private void setSystemHighAttributes() {
+    final Map<String, List<Serializable>> newSystemHighAttributes = new HashMap<>();
+
+    final Map<String, Object> systemHighUser =
+        (Map<String, Object>) json.get(SystemBaseUrl.getHost());
+    for (Map.Entry<String, Object> systemHighAttribute : systemHighUser.entrySet()) {
+      final String systemHighAttributeName = systemHighAttribute.getKey();
+
+      final Object systemHighAttributeValue = systemHighAttribute.getValue();
+      if (systemHighAttributeValue instanceof String) {
+        newSystemHighAttributes.put(
+            systemHighAttributeName, Collections.singletonList((String) systemHighAttributeValue));
+      } else {
+        newSystemHighAttributes.put(
+            systemHighAttributeName, (List<Serializable>) systemHighAttribute.getValue());
+      }
```

Comment:
❓  Should we be throwing an exception if the value is not one of the expected types? If we do, we should inline initialize `this.systemHighAttributes` with an empty map.

Reply:
❓ Or should we log an error and move on to the next attribute? Fail fast or best effort? Is it better to keep all the old values if something's wrong with the new map, or use as many new values as possible?
❗️ And we should also check the type before casting to `List<Serializable>` to avoid getting a `ClassCastException`.

Reply:
Regarding your second comment, the reason I chose to do it this way is the same as my other few replies. I want the initialization of AttributeFileClaimsHandler to fail (i.e. throw and exception) if there is an issue reading the file. In this case, the issue is that the values aren't serializable. Furthermore, a `ClassCastException` should never occur if the `users.attributes` file was set up according to the Documentation guidelines.

Reply:
In the most recent pushed changes, I updated this class to throw `IllegalStateException`s when the attributes file doesn't follow our assumptions.